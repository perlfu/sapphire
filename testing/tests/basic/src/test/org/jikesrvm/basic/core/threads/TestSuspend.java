/*
 * This file is part of Jikes RVM (http://jikesrvm.sourceforge.net).
 * The Jikes RVM project is distributed under the Common Public License (CPL).
 * A copy of the license is included in the distribution, and is also
 * available at http://www.opensource.org/licenses/cpl1.0.php
 *
 * (C) Copyright IBM Corp. 2001
 */
package test.org.jikesrvm.basic.core.threads;

/**
 * @author unascribed
 */
class TestSuspend extends XThread {

  static Thread sleeper;

  public static void main(String[] args) {
    XThread.holdMessages = false;
    sleeper = Thread.currentThread();
    new TestSuspend().start();
    XThread.say("suspending self");
    sleeper.suspend();
    XThread.say("resumed");
    XThread.say("bye");
    XThread.outputMessages();
  }

  public TestSuspend() {
    super("Resumer");
  }

  void performTask() {
    try {
      Thread.sleep(5000);
    } catch (Exception e) {
      e.printStackTrace();
    }
    XThread.say("resume sleeper...");
    try {
      sleeper.resume();
    } catch (Exception e) {
      XThread.say("error during resume: " + e);
      System.exit(1);
    }
  }
}
