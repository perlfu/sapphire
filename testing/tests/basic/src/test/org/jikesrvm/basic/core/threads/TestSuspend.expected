Thread 0 Main: suspending self
Thread 0 Main: resumed
Thread 0 Main: bye
Thread 1 Resumer: creating
Thread 1 Resumer: starting
Thread 1 Resumer: run starting
Thread 1 Resumer: resumming sleeper...
Thread 1 Resumer: run finishing
